CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contributing to the Openshift Builder
+
+The OpenShift Builder image drives [OpenShift Builds](https://docs.okd.io/latest/dev_guide/builds/index.html),
+leveraging [Buildah](https://github.com/containers/buildah) and [Source-To-Image](https://github.com/openshift/source-to-image)
+to turn source code into images that can be deployed on Kubernetes.
+
+## Getting the code
+
+To get started, [fork](https://help.github.com/articles/fork-a-repo) the [openshift/builder](https://github.com/openshift/builder) repo.
+
+## Developing
+
+### Testing on an OpenShift Cluster
+
+The easiest way to test your changes is to launch an OpenShift 4.x cluster.
+First, go to [try.openshift.com](https://try.openshift.com) to obtain a pull secret and download the installer.
+Follow the instructions to launch a cluster on AWS.
+
+If you want the latest OpenShift installer and `oc` clients, go to the [openshift-release](https://openshift-release.svc.ci.openshift.org/) 
+page, select the channel and build you wish to install, and download the respective `oc` and `openshift-installer` binaries.
+There are three types of channels you can obtain the installer from:
+
+1. `4-stable` - these are stable releases of OpenShift 4, corresponding to GA or beta releases.
+2. `4.x.0-nightly` - nightly development releases, with payloads published to quay.io.
+3. `4.x.0-ci` - bleeding-edge releases published to the OpenShift CI imagestreams.
+
+**Note**: Installs from the `4.x.0-ci` channel require a pull secret to `registry.svc.ci.openshift.org`, which is only available to Red Hat OpenShift developers.
+
+After your cluster is installed, you will need to do the following:
+
+1. Patch the cluster version so that you can launch your own builder image:
+
+```
+$ oc patch clusterversion/version --patch '{"spec":{"overrides":[{"kind":"ConfigMap", "namespace":"openshift-controller-manager-operator","name":"openshift-controller-manager-images","unmanaged":true}]}}' --type=merge
+```
+
+2. Make your code changes and build the binary with `make build`.
+3. Build the image using the `Dockerfile.dev` file, giving it a unique tag:
+
+```
+$ buildah bud -t <MYREPO>/<MYIMAGE>:<MYTAG> -f Dockerfile.dev .
```

Comment:
You can also use make build-devel-image here (which seems to use docker), might be worth noting, or update the Makefile?

Reply:
done